File: Kotlin/Entities.kt
package com.kamino.banking.entity

import jakarta.persistence.*
import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.*

@Entity
@Table(name = "extrato_banco")
class TransacaoFinanceira {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    var data: LocalDateTime = LocalDateTime.now()

    @Column(precision = 18, scale = 2)
    var valor: BigDecimal = BigDecimal.ZERO

    @Column(length = 200)
    var descricao: String? = null

    var dataHoraInclusao: LocalDateTime = LocalDateTime.now()

    var codigoNoBanco: String? = null

    var idContaBanco: Int = 0

    var idTransacaoExterna: UUID? = null

    val positivo: Boolean
        get() = valor >= BigDecimal.ZERO
}

@Entity
@Table(name = "conta_banco")
class ContaBanco {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0

    var idConfigAppExterno: Int? = null

    var descricao: String? = null
}

@Entity
@Table(name = "conta_pagar")
class ContaPagar {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0

    var valor: BigDecimal = BigDecimal.ZERO

    var pago: Boolean = false

    var dataPagamento: LocalDateTime? = null
}

File: TypeScript/entities.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('extrato_banco')
export class TransacaoFinanceira {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'datetime' })
  data: Date = new Date();

  @Column({ type: 'decimal', precision: 18, scale: 2 })
  valor: number = 0;

  @Column({ length: 200, nullable: true })
  descricao: string;

  @Column({ type: 'datetime' })
  dataHoraInclusao: Date = new Date();

  @Column({ nullable: true })
  codigoNoBanco: string;

  @Column()
  idContaBanco: number;

  @Column({ nullable: true })
  idTransacaoExterna: string;

  get positivo(): boolean {
    return this.valor >= 0;
  }
}

@Entity('conta_banco')
export class ContaBanco {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ nullable: true })
  idConfigAppExterno: number;

  @Column({ nullable: true })
  descricao: string;
}

@Entity('conta_pagar')
export class ContaPagar {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'decimal', precision: 18, scale: 2 })
  valor: number = 0;

  @Column({ default: false })
  pago: boolean = false;

  @Column({ type: 'datetime', nullable: true })
  dataPagamento: Date;
}
